### PR TITLE
Added Motr LNET Document

### PR DESCRIPTION
**Pull request by VenkyOS**
Signed-off-by: Venkataraman Padmanabhan <venkataraman.padmanabhan@seagate.com>

### CODE REVIEW

File: doc/Motr_LNet_Transport.rst [hunk 1]
```diff
@@ -0,0 +1,625 @@
+==============================
+HLD OF Motr LNet Transport
+==============================
+
+This document presents a high level design (HLD) of the Motr LNet Transport. The main purposes of this document are: (i) to be inspected by M0 architects and peer designers to ascertain that high level design is aligned with M0 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
```
Comment (huanghua78) [inline doc/Motr_LNet_Transport.rst:5]: I see you have replaced the old name with new "Motr".
Please also replace "M0" with "Motr" in this paragraph.

File: doc/Motr_LNet_Transport.rst [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+Introduction
+***************
+
+The scope of this HLD includes the net.lnet-user and net.lnet-kernel tasks described in [1]. Portions of the design are influenced by [4].
+
+***************
+Definitions
+***************
+
+- Network Buffer: This term is used to refer to a struct M0_net_buffer. The word “buffer”, if used by itself, will be qualified by its context - it may not always refer to a network buffer.
```
Comment (huanghua78) [inline doc/Motr_LNet_Transport.rst:17]: s/M0_net_buffer/m0_net_buffer/g

Please check and correct all of them.

Comment (huanghua78) [reply doc/Motr_LNet_Transport.rst:17]: It's better to have "Network Buffer" using the font **BOLD**, to match the original document.

So, to keep the document more readable, we need more beautiful page format.

File: doc/Motr_LNet_Transport.rst (hunk 1 above)
Comment (VenkyOS) [reply doc/Motr_LNet_Transport.rst]: Incorporated

File: doc/Motr_LNet_Transport.rst (hunk 2 above)
Comment (VenkyOS) [reply doc/Motr_LNet_Transport.rst]: Incorporated.

**Review by mukul-seagate11 — APPROVED** [review]
looks good

File: doc/Motr_LNet_Transport.rst [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+
+- **[r.m0.net.xprt.auto-provisioned-receive-buffer-pool]** Provide support for a pool of network buffers from which transfer machines can automatically be provisioned with receive buffers. Multiple transfer machines can share the same pool, but each transfer machine is only associated with a single pool. There can be multiple pools in a network domain, but a pool cannot span multiple network domains.
+
+******************
+Design Highlights
+******************
+
+The following figure shows the components of the proposed design and usage relationships between it and other related components:
+
+.. image:: Images/LNET.PNG
```
Comment (huanghua78) [inline doc/Motr_LNet_Transport.rst:89]: These two image still have "c2_" prefix words.
Can you change it to "m0_" in the image?

Comment (VenkyOS) [reply doc/Motr_LNet_Transport.rst:89]: Sure. I will work on that.

Comment (VenkyOS) [reply doc/Motr_LNet_Transport.rst]: Incorporated.

**Review by huanghua78 — APPROVED** [review]

**Comment by huanghua78** [comment]
The images in this files are not high resolution. Some texts are not shown.

But let's improve them in future.

**Comment by huanghua78** [comment]
@johnbent @nikita, we are going to remove the support of LNET transport.
Do you think this document is still needed?

**Comment by VenkyOS** [comment]
> The images in this files are not high resolution. Some texts are not shown.
> 
> But let's improve them in future.

Sure @huanghua78

**Comment by johnbent** [comment]
@huanghua78 , until we finalize the libfabric replacement, we still need this.  Also, this might remain as another network module for people in the community who might prefer it.  So I think we do still need this.

**Pull request merged**